name and qual name appear to be missing from decorated flask operation

### Description

I think `\_\_name\_\_` and `\_\_qualname\_\_` missing from FlaskOperator

I think it needs this

```
class FlaskOperation:
    def __init__(
        self,
        fn: t.Callable,
        jsonifier: Jsonifier,
        operation_id: str,
        pythonic_params: bool,
    ) -> None:
        # Something is acting like a wrapper but not copying these
        self.__name__ = fn.__name__
        self.__qualname__ = fn.__qualname__
        # end of suggestion
        self._fn = fn
```

### Expected behaviour

decorated functions should have function/callable metadata like name and qualname

### Actual behaviour

Missing such metatadata.

### Steps to reproduce

Install flask-limiter, decorate a controller function, call it, boom.  See stack trace below.

### Additional info:

```
❯ python --version
Python 3.11.1
❯ pip show connexion | grep "^Version\:"
Version: 3.0.dev0
```

#### error message
```
2023-03-05 10:25:31,498 [ERROR] search_service.uwsgi_config: HTTP Error
Traceback (most recent call last):
  File "E:\.virtualenv\search_service-QBGhH39p\Lib\site-packages\flask\app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\.virtualenv\search_service-QBGhH39p\Lib\site-packages\flask\app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\.virtualenv\search_service-QBGhH39p\Lib\site-packages\connexion\apps\flask.py", line 66, in __call__
    return self.fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\.virtualenv\search_service-QBGhH39p\Lib\site-packages\connexion\decorators\main.py", line 122, in wrapper
    return decorated_function(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\.virtualenv\search_service-QBGhH39p\Lib\site-packages\connexion\decorators\response.py", line 160, in wrapper
    handler_response = function(*args, **kwargs)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\.virtualenv\search_service-QBGhH39p\Lib\site-packages\connexion\decorators\parameter.py", line 84, in wrapper
    return function(**kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "E:\.virtualenv\search_service-QBGhH39p\Lib\site-packages\connexion\decorators\main.py", line 111, in wrapper
    return function(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\.virtualenv\search_service-QBGhH39p\Lib\site-packages\flask_limiter\extension.py", line 1178, in __inner
    if view_func and not get_qualified_name(view_func) == name:
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\.virtualenv\search_service-QBGhH39p\Lib\site-packages\flask_limiter\util.py", line 32, in get_qualified_name
    return f"{callable.__module__}.{callable.__name__}.{callable.__qualname__}"
                                    ^^^^^^^^^^^^^^^^^
AttributeError: 'FlaskOperation' object has no attribute '__name__'
```

--- a/connexion/apps/asynchronous.py
+++ b/connexion/apps/asynchronous.py
@@ -2,6 +2,7 @@
 This module defines a native connexion asynchronous application.
 """
 
+import functools
 import logging
 import pathlib
 import typing as t
@@ -24,19 +25,16 @@ logger = logging.getLogger(__name__)
 class AsyncOperation:
     def __init__(
         self,
-        operation: AbstractOperation,
         fn: t.Callable,
-        uri_parser: AbstractURIParser,
         jsonifier: Jsonifier,
         operation_id: str,
         pythonic_params: bool,
     ) -> None:
-        self._operation = operation
         self._fn = fn
-        self.uri_parser = uri_parser
         self.jsonifier = jsonifier
         self.operation_id = operation_id
         self.pythonic_params = pythonic_params
+        functools.update_wrapper(self, fn)
 
     @classmethod
     def from_operation(
@@ -47,9 +45,7 @@ class AsyncOperation:
         jsonifier: Jsonifier,
     ) -> "AsyncOperation":
         return cls(
-            operation,
-            fn=operation.function,
-            uri_parser=operation.uri_parser_class,
+            operation.function,
             jsonifier=jsonifier,
             operation_id=operation.operation_id,
             pythonic_params=pythonic_params,

--- a/connexion/apps/flask.py
+++ b/connexion/apps/flask.py
@@ -1,7 +1,7 @@
 """
 This module defines a FlaskApp, a Connexion application to wrap a Flask application.
 """
-
+import functools
 import pathlib
 import typing as t
 
@@ -36,6 +36,7 @@ class FlaskOperation:
         self.jsonifier = jsonifier
         self.operation_id = operation_id
         self.pythonic_params = pythonic_params
+        functools.update_wrapper(self, fn)
 
     @classmethod
     def from_operation(
@@ -104,7 +105,7 @@ class FlaskApi(AbstractRoutingAPI):
         return flask_path, endpoint_name
 
     def _add_operation_internal(
-        self, method: str, path: str, operation: FlaskOperation, name: str = None
+        self, method: str, path: str, operation: t.Callable, name: str = None
     ) -> None:
         self.blueprint.add_url_rule(path, name, operation, methods=[method])
 

